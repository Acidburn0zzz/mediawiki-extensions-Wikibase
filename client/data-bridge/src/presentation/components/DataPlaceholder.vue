<template>
	<p>
		Target Value: {{ targetValue }}
	</p>
</template>

<script lang="ts">
import {
	Component,
	Vue,
	Prop,
} from 'vue-property-decorator';

@Component
export default class DataPlaceholder extends Vue {
	@Prop( {
		required: true,
	} )
	public targetValue!: null|string;
}
</script>

<style scoped></style>
