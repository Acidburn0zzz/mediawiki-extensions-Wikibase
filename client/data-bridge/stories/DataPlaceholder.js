import { storiesOf } from '@storybook/vue';

import DataPlaceholder from '@/presentation/components/DataPlaceholder';

storiesOf( 'DataPlaceholder', module )
	.add( 'with example data', () => ( {
		components: { DataPlaceholder },
		template: '<DataPlaceholder target-value="example string"/>',
	} ) );
