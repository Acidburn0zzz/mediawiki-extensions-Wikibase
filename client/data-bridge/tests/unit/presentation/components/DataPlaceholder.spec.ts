import DataPlaceholder from '@/presentation/components/DataPlaceholder.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'DataPlaceholder', () => {
	it( 'renders correctly', () => {
		const wrapper = shallowMount( DataPlaceholder, {
			propsData: {
				targetValue: 'some string',
			},
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );
} );
