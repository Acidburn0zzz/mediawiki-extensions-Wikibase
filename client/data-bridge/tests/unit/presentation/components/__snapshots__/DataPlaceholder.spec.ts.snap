// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPlaceholder renders correctly 1`] = `
<p>
  
	Target Value: some string

</p>
`;
